Bottom: 251d6c3eddbbecaffe60b46c30a70d7cc633c184
Top:    2522cd720f5432548e55a4a175b9a4d99c2904aa
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-01 13:22:39 +0100

toporouter.c: Re-implement coord_move_towards_coord_values() with atan2


---

diff --git a/src/toporouter.c b/src/toporouter.c
index 61d94d5..fa2ce36 100644
--- a/src/toporouter.c
+++ b/src/toporouter.c
@@ -783,44 +783,16 @@ sloppy_vertex_wind(GtsVertex *a, GtsVertex *b, GtsVertex *c)
 }
 
 /* moves vertex v d units in the direction of vertex p */
-void
-coord_move_towards_coord_values(gdouble ax, gdouble ay, gdouble px, gdouble py, gdouble d, gdouble *x, gdouble *y) 
+static void
+coord_move_towards_coord_values (double ax, double ay,
+                                 double px, double py,
+                                 double d,
+                                 double *x, double *y)
 {
-  gdouble dx = px - ax;
-  gdouble dy = py - ay;
-  gdouble theta = atan(fabs(dy/dx));
-
-
-  if(!finite(theta)) {
-    printf("!finite(theta) a = %f,%f p = %f,%f d = %f\n", 
-        ax, ay, px, py, d);
-
-  }
-
-  g_assert(finite(theta));
-
-  if( dx >= 0. ) {
-
-    if( dy >= 0. ) {
-      *x = ax + (d * cos(theta));
-      *y = ay + (d * sin(theta));
-    }else{
-      *x = ax + (d * cos(theta));
-      *y = ay - (d * sin(theta));
-    }
-
-  }else{
-    
-    if( dy >= 0. ) {
-      *x = ax - (d * cos(theta));
-      *y = ay + (d * sin(theta));
-    }else{
-      *x = ax - (d * cos(theta));
-      *y = ay - (d * sin(theta));
-    }
-
-  }
+  double theta = atan2 (py - ay, px - ax);
 
+  *x = ax + d * cos (theta);
+  *y = ay + d * sin (theta);
 }
 
 /* moves vertex v d units in the direction of vertex p */
